Automattic/jetpack
# Map block: Allow block selection when Mapkit is being used

Fixes https://github.com/Automattic/jetpack/issues/31870

## Proposed changes:

* This PR adds an overlay div to the Mapkit map markup that is rendered, in order to allow the block to be selectable.
* This fixes an issue where it isn't possible to use a mouse to select the block just by clicking on the block, within the wp-admin post / page editor.

### Other information:


## Jetpack product discussion

https://github.com/Automattic/jetpack/issues/31870

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

On self-hosted:

* Though this isn't an issue for self-hosted sites as they use Mapbox for the maps used in the Map block, you can still replicate the issue and test the fix there. 
* When adding a Map block to a post or page, add the following via the developer console: `document.cookie = "map_provider=mapkit";`. You should then be able to add a map using Mapkit. Make sure the map style is not 'Terrain' as that uses Mapbox.
* On trunk, notice that you cannot use your mouse to select the block (the map block settings are not visible in the sidebar, you can only move the map).
* With this PR applied, you will be able to see the map settings in the sidebar. With an extra mouse click, you can also move around the map as before.

On WoA / Simple:
* Add a Map block to a post or page in wp-admin, not WordPress.com (eg. `/wp-admin/post.php?post={postnumber}&action=edit`)., and without the PR applied you should be able to replicate the issue as above. Make sure the map style is not 'Terrain' as that uses Mapbox.
* On WoA apply the PR via the beta tester plugin, and on Simple you can use the command in the generated comment below to apply the changes to a sandboxed site.
* You should now be able to select the Map block itself and see the map settings, just by clicking on the map. An additional click should allow you to move around the map.
